torch/_decomp/decompositions.py

```diff
@@ -3434,6 +3434,103 @@ def arange_start(
     )
 
 
+def _multi_margin_loss_shape_check(ndims, input, target):
+    valid_inputs = (
+        (ndims == 2 and input.size(1) != 0)
+        or (ndims == 1 and input.size(0) != 0)
+        or ndims == 0
+    )
+    if ndims <= 1:
+        nframe = 1
+        dim = 1 if ndims == 0 else input.size(0)
+    else:
+        nframe = input.size(0)
+        dim = input.size(1)
+    torch._check(
+        valid_inputs,
+        lambda: f"Expected non-empty vector or matrix with optional 0-dim batch size, but got: {input.shape}",
+    )
+    torch._check(
+        valid_inputs and target.ndim <= 1 and target.numel() == nframe,
+        lambda: f"inconsistent target size, got: {target.shape}",
+    )
+    return nframe, dim
+
+
+@register_decomposition(aten.multi_margin_loss)
+@out_wrapper()
+def multi_margin_loss(
+    input: Tensor,
+    target: Tensor,
+    p: NumberType = 1,
+    margin: NumberType = 1,
+    weight: Optional[Tensor] = None,
+    reduction: int = Reduction.MEAN.value,
+) -> Tensor:
+    ndims = input.ndim
+    torch._check(p == 1 or p == 2, lambda: "only p == 1 and p == 2 supported")
+    _multi_margin_loss_shape_check(ndims, input, target)
+    input = torch.atleast_2d(input)
+    target = torch.atleast_1d(target)
+    if weight is not None:
+        weight = torch.atleast_1d(weight)
+    size = input.size(1)
+    mask = torch.arange(size, device=input.device) != target.unsqueeze(1)
```

Comment:
This part is not documented either? Could you please update the doc accordingly in another PR?

Reply:
Which part is "not documented"? Not sure what you're referring to.

Here, I exclude target values from some computations. In the doc, there's an example at the very bottom: https://pytorch.org/docs/stable/generated/torch.nn.MultiMarginLoss.html

Or do you mean just add an example for target values when used with batches?

Reply:
oh, it does say `i != y` in the docs, nevermind. It's just that the docs are not particularly good.